payment/src/main/java/org/killbill/billing/payment/core/sm/PaymentAutomatonRunner.java

```diff
@@ -128,8 +129,10 @@ public PaymentStateContext buildPaymentStateContext(final boolean isApiPayment,
                                                         final Iterable<PluginProperty> properties,
                                                         final CallContext callContext,
                                                         final InternalCallContext internalCallContext) throws PaymentApiException {
+
         // Retrieve the payment id from the payment external key if needed
         final UUID effectivePaymentId = paymentId != null ? paymentId : retrievePaymentId(paymentExternalKey, paymentTransactionExternalKey, internalCallContext);
+        verifyPaymentMethodExistsForRefund(effectivePaymentId, transactionType, internalCallContext);
```

Comment:
> Is that correct? Can a payment transaction be created over a deleted payment method?

Chargebacks are special... 😼 

One scenario would be the following:

* Shopper signs-up for free trial
* 30 days later, the shopper is auto-enrolled into a paying plan. CC is charged.
* When the shopper looks at his statement, he forgot signing-up for the service and canceling the free trial. He goes on deleting his account including the payment method.
* Instead of trying to get a refund with customer support, he issues a chargeback with the CC company. Kill Bill might get notified of the chargeback [months](https://chargebacks911.com/chargeback-time-limit/) after the shopper has closed his account. At this point, the payment method has been deleted but we still need to record that chargeback for accounting purposes.

All other operations (auth, capture, void, refund, credit, purchase) do require a valid payment method on file though.